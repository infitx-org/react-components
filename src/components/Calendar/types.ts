export type PossibleDate = Date | undefined;

export type PossibleDay = string | undefined;

export type Matrix = PossibleDay[][];

export type DisabledDays = (day: Date) => boolean;

export type DateRange = [PossibleDate, PossibleDate];
