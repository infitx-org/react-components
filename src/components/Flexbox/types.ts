export enum TopBottom {
  "top" = "flex-start",
  "bottom" = "flex-end",
}
export enum LeftRight {
  "left" = "flex-start",
  "right" = "flex-end",
}

export enum Mappers {
  "top" = "flex-start",
  "bottom" = "flex-end",
  "left" = "flex-start",
  "right" = "flex-end",
}

export type AlignItems =
  | "flex-start"
  | "flex-end"
  | "center"
  | "baseline"
  | "stretch";

export type JustifyContent =
  | "flex-start"
  | "flex-end"
  | "center"
  | "space-between"
  | "space-around"
  | "space-evenly";

export type AlignWithMap<T extends string> = `${AlignItems}` | `${T}`;
export type JustifyWithMap<T extends string> = `${JustifyContent}` | `${T}`;
