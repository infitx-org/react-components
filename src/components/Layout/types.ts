export enum AlignMap {
  "top" = "flex-start",
  "bottom" = "flex-end",
}

export enum JustifyMap {
  "left" = "flex-start",
  "right" = "flex-end",
}

export type AlignItems =
  | "flex-start"
  | "flex-end"
  | "center"
  | "baseline"
  | "stretch";

export type JustifyContent =
  | "flex-start"
  | "flex-end"
  | "center"
  | "space-between"
  | "space-around"
  | "space-evenly";

export type AlignWithMap = `${AlignItems | keyof typeof AlignMap}`;
export type JustifyWithMap = `${JustifyContent | keyof typeof JustifyMap}`;
