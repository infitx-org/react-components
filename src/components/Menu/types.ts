export interface MenuSectionProps {
  label: string;
  hidden?: boolean;
  icon?: React.ReactElement<React.SVGProps<SVGSVGElement>>;
  fill?: string;
  size?: number;
  disabled?: boolean;
  children: MenuItemElement | MenuItemElement[];
}

interface BaseMenuItemProps {
  label: string;
  active?: boolean;
  disabled?: boolean;
  hidden?: boolean;
  back?: boolean;
  icon?: React.ReactElement<React.SVGProps<SVGSVGElement>>;
  fill?: string;
  size?: number;
  // prop partial is not used directly by the MenuItem but it is used by the Menu component
  partial?: boolean;
  children?: MenuItemElement | MenuElement[];
}

interface BaseMenuItemPropsWithTo extends BaseMenuItemProps {
  to: string;
  path?: string;
}
interface BaseMenuItemPropsWithPath extends BaseMenuItemProps {
  to?: string;
  path: string;
}

export type MenuItemProps = BaseMenuItemPropsWithTo | BaseMenuItemPropsWithPath;

export type MenuItemElement = React.ReactElement<MenuItemProps>;
export type MenuSectionElement = React.ReactElement<MenuSectionProps>;
export type MenuElement = MenuItemElement | MenuSectionElement;
