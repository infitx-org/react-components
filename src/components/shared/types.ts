export type BaseInput = Omit<
  React.InputHTMLAttributes<HTMLInputElement>,
  "size" | "onChange"
>;

export type BaseSelect = Omit<
  React.SelectHTMLAttributes<HTMLInputElement>,
  "size" | "value" | "onChange"
>;
