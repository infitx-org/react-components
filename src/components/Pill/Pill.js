import './Pill.scss';

import { composeClassNames } from 'common/generic';
import PropTypes from 'prop-types';
import React from 'react';

import Icon from '../Icon';
import Tooltip from '../Tooltip';

const Pill = ({ id, className, kind = 'default', icon, fill, label, active, tooltip }) => {
  const pillClassName = composeClassNames([
    'rc-pill',
    `rc-pill--${kind}`,
    active ? `rc-pill--active` : `rc-pill--inactive`,
    className,
  ]);

  let pill = (
    <div className={pillClassName} id={id}>
      {icon && (
        <div className="rc-pill__icon-box">
          <Icon name={icon} size={16} className="rc-pill__icon" fill={fill} />
        </div>
      )}
      {label && <span className="rc-pill__label">{label}</span>}
    </div>
  );

  if (tooltip) {
    pill = <Tooltip label={tooltip}>{pill}</Tooltip>;
  }
  return pill;
};

export default Pill;

Pill.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  kind: PropTypes.oneOf([
    'default',
    'primary',
    'secondary',
    'tertiary',
    'success',
    'danger',
    'warning',
    'dark',
    'light',
  ]),
  icon: PropTypes.string,
  fill: PropTypes.string,
  label: PropTypes.string,
  active: PropTypes.bool,
  tooltip: PropTypes.string,
};
