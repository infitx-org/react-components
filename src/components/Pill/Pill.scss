@use "sass:map";
@import "../../scss/base/colors.scss";
@import "../../scss/base/shadow.scss";
@import "../../scss/themes.scss";
@import "../../scss/mixins.scss";

.rc-pill {
  @include rc-component;
  border-radius: 20px;
  border: 2px solid;
  min-height: 24px;
  padding: 3px 7px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  cursor: initial;
  border-color: transparent;
}

.rc-pill--default {
  fill: #333;
  color: #333;
  background-color: #eee;

  &.rc-pill--inverted {
    fill: #eee;
    color: #eee;
    background-color: #333;
  }
}

@each $kind in $kinds {
  $base-color: map-get($theme-colors, $kind);
  $light-color: lighten(map-get($theme-shadows, $kind), 10%);

  .rc-pill--#{$kind} {
    fill: $base-color;
    color: $base-color;
    background-color: $light-color;

    &.rc-pill--inverted {
      fill: $light-color;
      color: $light-color;
      background-color: $base-color;
    }
  }
}

.rc-pill--active {
  border-color: currentColor;
}

.rc-pill__icon-box {
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0px 2px;
}

.rc-pill__label {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  font-size: 12px;
  font-weight: bold;
  margin: 0px 2px;
  cursor: initial;
}
