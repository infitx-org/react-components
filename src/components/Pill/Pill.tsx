import classnames from "classnames";
import { Kind } from "types";
import Icon from "components/Icon";
import "./Pill.scss";

interface PillProps {
  id?: string;
  className?: string;
  style?: React.CSSProperties;
  kind?: `${Kind}` | "default";
  icon?: React.ReactElement<React.SVGProps<SVGSVGElement>>;
  fill?: string;
  label?: string;
  active?: boolean;
  inverted?: boolean;
}

function Pill({
  id,
  className,
  style,
  kind = "default",
  icon,
  fill = "inherit",
  label,
  active,
  inverted,
}: PillProps) {
  const pillClassName = classnames([
    "rc-pill",
    `rc-pill--${kind}`,
    active && `rc-pill--active`,
    inverted && `rc-pill--inverted`,
    className,
  ]);

  return (
    <div className={pillClassName} id={id} style={style}>
      {icon && (
        <div className="rc-pill__icon-box">
          <Icon icon={icon} size={16} className="rc-pill__icon" fill={fill} />
        </div>
      )}
      {label && <span className="rc-pill__label">{label}</span>}
    </div>
  );
}

export default Pill;
