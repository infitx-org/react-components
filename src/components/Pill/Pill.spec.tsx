import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { Kind } from "types";
import Pill from "./Pill";

const icon = (
  <svg viewBox="0 0 20 20">
    <circle cx="10" r="5" cy="10" />
  </svg>
);

describe("test the Pill component", () => {
  it("renders the component", () => {
    const { container } = render(<Pill />);
    expect(container.querySelector(".rc-pill")).toBeTruthy();
  });

  it("renders the label", () => {
    const { container } = render(<Pill label="Hello" />);
    expect(container.querySelector(".rc-pill__label")).toBeTruthy();
    expect(container.querySelector(".rc-pill__label")).toHaveTextContent(
      "Hello"
    );
  });

  it("renders the active state", () => {
    const { container } = render(<Pill active />);
    expect(container.querySelector(".rc-pill")).toHaveClass("rc-pill--active");
  });

  it("renders the inverted state", () => {
    const { container } = render(<Pill inverted />);
    expect(container.querySelector(".rc-pill")).toHaveClass(
      "rc-pill--inverted"
    );
  });

  it("renders the icon", () => {
    const { container } = render(<Pill icon={icon} />);
    expect(container.querySelector(".rc-pill__icon")).toBeTruthy();
  });

  it("renders the prop fill", () => {
    const { container } = render(<Pill icon={icon} fill="#fff" />);
    expect(container.querySelector("svg")).toHaveAttribute("fill", "#fff");
  });

  it("renders the prop className", () => {
    const { container } = render(<Pill className="test" />);
    expect(container.querySelector(".rc-pill")).toHaveClass("test");
  });

  it("renders the prop id", () => {
    const { container } = render(<Pill id="testPillId" />);
    expect(container.querySelector(".rc-pill")).toHaveAttribute(
      "id",
      "testPillId"
    );
  });

  it("renders the default correct kind", () => {
    const { container } = render(<Pill icon={icon} />);
    expect(container.querySelector(".rc-pill")).toHaveClass("rc-pill--default");
  });

  it("renders all the kinds", () => {
    Object.values(Kind).forEach((kind) => {
      const { container } = render(<Pill kind={kind} />);
      expect(container.querySelector(".rc-pill")).toHaveClass(
        `rc-pill--${kind}`
      );
    });
  });

  it("renders the component correctly when multiple props are set", () => {
    const { container } = render(
      <Pill icon={icon} kind="secondary" active fill="#fff" />
    );
    expect(container).toMatchSnapshot();
  });
});
