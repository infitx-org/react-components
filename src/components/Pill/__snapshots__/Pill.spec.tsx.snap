// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the Pill component renders the component correctly when multiple props are set 1`] = `
<div>
  <div
    class="rc-pill rc-pill--secondary rc-pill--active"
  >
    <div
      class="rc-pill__icon-box"
    >
      <svg
        class="rc-pill__icon"
        fill="#fff"
        height="16"
        stroke="transparent"
        viewBox="0 0 20 20"
        width="16"
      >
        <circle
          cx="10"
          cy="10"
          r="5"
        />
      </svg>
    </div>
  </div>
</div>
`;
